resharper/test/data/CSharp/Daemon/Stages/Analysis/DrawGizmoAttributeProblemAnalyzer.cs

```diff
@@ -0,0 +1,30 @@
+﻿using UnityEngine;
+using UnityEditor;
+
+public class TestDrawGizmoMethod
+{
+    [DrawGizmo]
+    public void NotStatic(Component arg1, GizmoType arg2) { }
+
+    [DrawGizmo]
+    public static int WrongReturnType(Component arg1, GizmoType arg2) { }
+
+    [DrawGizmo]
+    public static void WrongReturnType(Component arg1, GizmoType arg2) { }
+
+    [DrawGizmo]
+    public static void OneCorrectParameter(GameObject arg1, GizmoType arg2) { }
+	
+    [DrawGizmo]
+    public static void OneParameter(GizmoType arg2) { }
+	
+    [DrawGizmo]
+    public static void WrongTypeParameters<T>(Component arg1, GizmoType arg2) { }
+
+    [DrawGizmo]
+    public static void JustRight(Component arg1, GizmoType arg2) { }
+
+    [DrawGizmo]
+    private static void JustRight2(MonoBehaviour arg1, GizmoType arg2) { }
+}
```

Comment:
Can you add a couple more tests where the first parameter are derived types, please? Just add a couple of dummy classes into this file and derive from `MonoBehaviour` and `Component`.